tests/test_spatial_analysis/test_calculate_spatial_distance.py

```diff
@@ -0,0 +1,200 @@
+        self.adata.obs['cell_type'] = np.random.choice(
+            ['type1', 'type2'], size=100
+        )
+        self.adata.obs['imageid'] = np.random.choice(
+            ['image1', 'image2'], size=100
+        )
+
+    def test_typical_case(self):
+        """Test the function with default parameters."""
+        calculate_spatial_distance(
+            adata=self.adata,
+            annotation='cell_type',
+            verbose=False
+        )
+        self.assertIn('spatial_distance', self.adata.uns)
+
+    def test_output_structure(self):
+        """Test that the output is stored as a pandas DataFrame."""
+        calculate_spatial_distance(
+            adata=self.adata,
+            annotation='cell_type',
+            verbose=False
+        )
+        result = self.adata.uns['spatial_distance']
+        self.assertIsInstance(result, pd.DataFrame)
+
+    def test_subset_processing(self):
+        """Test that the function correctly processes a subset of images."""
+        # Run the function with subset=['image1']
+        calculate_spatial_distance(
+            adata=self.adata,
+            annotation='cell_type',
+            subset=['image1'],
+            imageid='imageid',
+            verbose=False
+        )
+        result = self.adata.uns['spatial_distance']
+
+        # Get all cell IDs in 'image1' as strings
+        image1_cell_indices = self.adata.obs[
+            self.adata.obs['imageid'] == 'image1'
+        ].index.astype(str)
+
+        # Get the cell indices in the result
+        result_cell_indices = result.index.astype(str)
+
+        # Check that all cell IDs in result are from 'image1'
+        self.assertTrue(
+            set(result_cell_indices).issubset(set(image1_cell_indices))
+        )
+
+    def test_invalid_subset_type(self):
+        """Test that passing an invalid subset type raises a TypeError."""
+        # Pass an integer as subset, which should raise TypeError
+        with self.assertRaises(TypeError):
+            calculate_spatial_distance(
+                adata=self.adata,
+                annotation='cell_type',
+                subset=123,  # Invalid subset type
+                imageid='imageid',
+                verbose=False
+            )
+
+    def test_subset_imageid_not_found(self):
+        """Test that passing a subset with non-existent image IDs
+        raises a ValueError."""
+        # Pass a subset with an image ID not present in the data
+        with self.assertRaises(ValueError):
+            calculate_spatial_distance(
+                adata=self.adata,
+                annotation='cell_type',
+                subset=['nonexistent_image'],
+                imageid='imageid',
+                verbose=False
+            )
+
+    def test_missing_coordinates(self):
+        """Test that the function raises a KeyError
+        when spatial coordinates are missing."""
+        # Remove the spatial coordinates
+        del self.adata.obsm['spatial']
```

Comment:
@fangliu117 , once you implement "check_associated_table", we won't need this test.

Reply:
Yes, removed this test.